c7n/utils.py

```diff
@@ -45,6 +45,13 @@
     except ImportError:  # pragma: no cover
         from yaml import SafeLoader, SafeDumper as BaseSafeDumper
 
+try:
+    # python 3
+    from urllib.request import getproxies
+except ImportError:
+    # python 2
+    from urllib import getproxies
```

Comment:
I think we should be using `six.moves.urllib.request` here instead of this way of doing it (assuming it exists).  We are doing that above for urlparse already.